woocommerce/woocommerce-ios
# Orders tab: fix order details shown instead of order list in some iPhone devices on iOS 17.0/17.1

Closes: #11709
Closes: #11675



## Why

When split view feature flag is enabled, upon opening the orders tab for the first time, order details are shown instead of the order list in the orders tab in some iPhone devices on iOS 17.0/17.1. It's because `UISplitViewController.isCollapsed` is currently used to check if split view is supported in the UI for an order to be selected, but the value can be `false` in `viewWillAppear` in compact devices on certain iOS versions like iOS 17.0.1 and 17.1.

## Changes

There's a [similar topic in SO](https://stackoverflow.com/questions/44336916/why-sometimes-iscollapsed-of-uisplitviewcontroller-is-false-on-iphone-portrait), and after some testing `UISplitViewController.isCollapsed` becomes `true` in iPhone devices after the first `viewDidLayoutSubviews`. This PR moves the check for selecting an order in split view from `viewWillAppear` to `viewDidLayoutSubviews` once, as `viewDidLayoutSubviews` can be called more than once.

## Testing instructions


Hand-testing is recommended. Please try reproducing the issue first.

- In code, enable the `splitViewInOrdersTab` feature flag in `DefaultFeatureFlagService` (returning `true`)
- Launch the app in iOS 17.0.1 or 17.1 simulator/device
- Switch to a store with at least one order, if needed
- Go to the orders tab --> the order list should be shown

### Tablet testing

- Launch the app on a tablet
- Go to the orders tab --> order details should be shown on the right pane while the order list is on the left


## Screenshots




https://github.com/woocommerce/woocommerce-ios/assets/1945542/193cff73-9a4b-414d-a890-bafeece9e7ea

https://github.com/woocommerce/woocommerce-ios/assets/1945542/819a4454-8135-47c8-a6e6-50c85a14c4ce



---
